File: cpp/cpp_1000-/2390.cc
// C++
// leetcode 2390
// https://leetcode.cn/problems/removing-stars-from-a-string/

class Solution {
public:
    string removeStars(string s) {
        int n = s.size(), p = 0;
        vector<char> tmp(n);
        for (int i = 0; i < n; i++) {
        	if (s[i] == '*') {
        		p--;
        	} else {
        		tmp[p++] =  s[i];
        	}
        }
        if (p == 0)
        	return "";
        tmp.resize(p);
        return string(tmp.begin(), tmp.end());
    }
};

File: Go/Go_1000-/2390.go
// Go
// leetcode 2390
// https://leetcode.cn/problems/removing-stars-from-a-string/

func removeStars(s string) string {
	n := len(s)
	tmp := make([]byte, n)
	var p int
	for i := 0; i < n; i++ {
		if s[i] == '*' {
			p--
		} else {
			tmp[p] = s[i]
			p++
		}
	}
	return string(tmp[0:p])
}